libraries/AP_Scripting/applets/VTOL-expo.lua

```diff
@@ -0,0 +1,287 @@
+--[[
+   estimation of MOT_THST_EXPO for VTOL thrust linearisation
+
+   should be used in copter GUIDED mode
+--]]
+
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+
+local PARAM_TABLE_KEY = 11
+local PARAM_TABLE_PREFIX = "EXPO_"
+
+local UPDATE_RATE_HZ = 100
+
+local GRAVITY_MSS = 9.80665
+
+-- bind a parameter to a variable
+function bind_param(name)
+   local p = Parameter()
+   assert(p:init(name), string.format('could not find %s parameter', name))
+   return p
+end
+
+-- add a parameter and bind it to a variable
+function bind_add_param(name, idx, default_value)
+   assert(param:add_param(PARAM_TABLE_KEY, idx, name, default_value), string.format('could not add param %s', name))
+   return bind_param(PARAM_TABLE_PREFIX .. name)
+end
+
+-- setup expo specific parameters
+assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 12), 'could not add param table')
+
+--[[
+  // @Param: EXPO_ENABLE
+  // @DisplayName: Expo  enable
+  // @Description: Enable expo tuning system
+  // @Values: 0:Disabled,1:Enabled
+  // @User: Standard
+--]]
+local EXPO_ENABLE      = bind_add_param('ENABLE',         1, 0)
+
+--[[
+  // @Param: EXPO_RC_FUNC
+  // @DisplayName: Expo RC function
+  // @Description: RCn_OPTION number to use to control tuning stop/start/save
+  // @User: Standard
+--]]
+local EXPO_RC_FUNC     = bind_add_param('RC_FUNC',        2, 300)
+
+--[[
+  // @Param: EXPO_PULSE_TIME
+  // @DisplayName: Expo pulse time
+  // @Description: Pulse length for expo measurement
+  // @Range: 0.1 1
+  // @User: Standard
+--]]
+local EXPO_PULSE_TIME  = bind_add_param('PULSE_TIME',     3, 0.25)
+
+--[[
+  // @Param: EXPO_LIMIT_MIN
+  // @DisplayName: Expo min value
+  // @Description: Minimum value we will tune to
+  // @Range: -1 1
+  // @User: Standard
+--]]
+local EXPO_LIMIT_MIN  = bind_add_param('LIMIT_MIN',       4, 0.0)
+
+--[[
+  // @Param: EXPO_LIMIT_MAX
+  // @DisplayName: Expo max value
+  // @Description: Maximum value we will tune to
+  // @Range: -1 1
+  // @User: Standard
+--]]
+local EXPO_LIMIT_MAX  = bind_add_param('LIMIT_MAX',       5, 0.9)
+
+-- get time in seconds since boot
+function get_time()
+   return millis():tofloat() * 0.001
+end
+
+local MOT_THST_EXPO = bind_param('MOT_THST_EXPO')
+local GUID_TIMEOUT = bind_param('GUID_TIMEOUT')
```

Comment:
```suggestion
local MOT_THST_EXPO = Parameter('MOT_THST_EXPO')
local GUID_TIMEOUT = Parameter('GUID_TIMEOUT')
```
We don't need the helper anymore.

Reply:
changed